src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessible.java

```diff
@@ -121,7 +122,11 @@ public void propertyChange(PropertyChangeEvent e) {
                 if (name.equals(ACCESSIBLE_CARET_PROPERTY)) {
                     selectedTextChanged(ptr);
                 } else if (name.equals(ACCESSIBLE_TEXT_PROPERTY)) {
-                    valueChanged(ptr);
+                    AccessibleContext thisAC = accessible.getAccessibleContext();
+                    Accessible parentAccessible = thisAC.getAccessibleParent();
+                    if (!(parentAccessible instanceof JSpinner.NumberEditor)) {
```

Comment:
Why do you need all this if the spinner has an accessible role?
```jdk-21.0.6+7/src/java.desktop/share/classes/javax/swing/JSpinner.java```
```JAVA
        public AccessibleRole getAccessibleRole() {
            return AccessibleRole.SPIN_BOX;
        }
```
Also, there is already getting an a11y parent lower in the code.

Reply:
Native spinner and Java spinner is not a similar UI component and the issue lies in the text component of JSpinner. I don't find any probable solution in native side implementation of spin box.

This seems a simple workaround fix for the issue right now.

Reply:
I like this solution, I just ask you to consider replacing the ```instance of``` check with a role comparison…

Reply:
ohh ok... role comparison will not work here because the role is "AccessibleRole.TEXT" and we want to ignore the property change only if the parent is a JSpinner with NumberEditor.

Reply:
We can check the role of the parent and the current element.
The parent should be a spin box, the current one should be a text element.

Reply:
I checked it before, the parent role is `Panel` and current element role is `Text` and that results in inconsistent announcement.